src/EDMF/Models/entr_detr.jl

```diff
@@ -19,6 +29,60 @@ Define entrainment and detrainment fields
 """
 function compute_entrainment_detrainment! end
 
+function compute_entrainment_detrainment!(grid::Grid{FT}, UpdVar, tmp, q, params, model::RH_Diff) where FT
+  gm, en, ud, sd, al = allcombinations(q)
+  Δzi = grid.Δzi
+  k_1 = first_interior(grid, Zmin())
+  @inbounds for i in ud
+    @inbounds for k in over_elems_real(grid)
+      # get parameters
+      @unpack params param_set
+      c_ε = model.entr_factor
+      c_turb = model.turb_entr_fac
+      if tmp[:q_liq, k, en]+tmp[:q_liq, k, i]>0.0
+        c_δ = model.detr_factor
+      else
+        c_δ = 0.0
+      end
+      β = model.detr_RH_power
+      μ_0 = model.sigmoid_slope_param
+      χ = model.upd_mixing_frac
+      c_λ = model.entr_tke_fac
+
+      # get subdomain properties
+      b_up = tmp[:buoy, k, i]
+      b_en = tmp[:buoy, k, en]
+      w_up = max(q[:w, k, i],1e-4)
+      w_en = q[:w, k, en]
+      dw = max(w_up - w_en,1e-4)
+      db = b_up - b_en
+      sqrt_tke = sqrt(max(q[:tke, k, gm],0.0))
+      ts = ActiveThermoState(param_set, q, tmp, k, en)
+      RH_en = relative_humidity(ts)
+      ts = ActiveThermoState(param_set, q, tmp, k, i)
+      RH_up = relative_humidity(ts)
+
+      # compute aux functions
+      D_ϵ = 1.0/(1.0+exp(-db/dw/μ_0*(χ - q[:a, k, i]/(q[:a, k, i]+q[:a, k, en]))))
```

Comment:
```suggestion
      D_ϵ = 1/(1+exp(-db/dw/μ_0*(χ - q[:a, k, i]/(q[:a, k, i]+q[:a, k, en]))))
```
`1.0` defaults to `Float32`, and is therefore type-unstable for `Float32`.

Reply:
Please follow this rule elsewhere